### ISSUE TITLE
InputPicker with virtualization select elements only after double clicks

### ISSUE THREAD

**Issue by sergeythrees** [rsuite/rsuite#3245]
### What version of rsuite are you using?

5.35.1

### What version of React are you using?

17.0.2

### What version of TypeScript are you using (if any)?

5.0.3

### What browser are you using?

Chrome

### Describe the Bug

InputPicker with virtualization item selects only after double clicks. 

[Form-Builder (3).webm](https://github.com/rsuite/rsuite/assets/22077517/f468b4dc-458a-4fd6-b918-2cbd29a90050)

### Expected Behavior

Without virtualization item selects by single click.
In SelectPicker with virtualization it works correct.

### To Reproduce

```js
  const listProps: Partial<ListProps> = {
    onItemsRendered: ({visibleStopIndex}) => {
      if (visibleStopIndex >= data.length - 1) {
        load()
      }
    }
  }

  const renderMenu = (menu: ReactNode) => <>
    {menu}
    {loading && <Loader/>}
  </>

  return (
      <InputPicker {...props} data={data} listProps={listProps} renderMenu={renderMenu} virtualized/>
  )
```

**Comment by SevenOutman** [comment]
I'm routing this issue to @simonguo as it's related to virtualization.

**Comment by shu8** [comment]
I'm also facing this issue with a `virtualized` `TagPicker`.

I found that the double click doesn't seem to actually need to be on the _same_ element. e.g., if you click on `element 1` then `element 2`, then `element 2` is what is finally checked. In fact it seems like the first click can be anywhere within the dropdown menu (e.g., any white space inside it).

This makes me wonder if the first click is acting as some sort of 'focus' event first?

I tried digging into the code but couldn't figure out exactly what's happening unfortunately. It seems like internally, `TagPicker` uses `InputPicker` which uses `DropdownMenu`. When I try a plain `DropdownMenu` that is `virtualized`, I don't need to double click. But when I use a `TagPicker`, I do need to double click.

Works (no need for double click):
```
<DropdownMenu
      dropdownMenuItemAs={DropdownMenuCheckItem}
      classPrefix="test"
      data={[{ label: "test", value: "test" }]}
      virtualized
      onSelect={(e) => console.log(e)}
/>
```

Doesn't work (requires double click):
```
<TagPicker
      data={[{ label: "test", value: "test" }]}
      virtualized
      onSelect={(e) => console.log(e)}
/>
```

I hope this can help in pinpointing the issue @simonguo. I'm happy to help write a fix if you are able to point me in the right direction! 🙂

**Comment by sergeythrees** [comment]
@SevenOutman @simonguo 
excuse me, when is it planned to fix this bug approximately?  Or is it a low priority?

**Comment by sergeythrees** [comment]
Reproduce:
https://codesandbox.io/s/8thpnd?file=/index.js:0-1398

**Comment by sergeythrees** [comment]
@simonguo @SevenOutman 
I found the cause of the bug and how to fix it . Please explain to me how to make a pul request. I haven't done pull requests to your repository yet.

**Comment by SevenOutman** [comment]
> I found the cause of the bug and how to fix it . Please explain to me how to make a pul request. I haven't done pull requests to your repository yet.

Thanks for being willing to help! To open a pull request, you would need to fork this repo, make changes in your own fork, then open a pull request targeting this repo.

As of how to develop this project locally, you could refer to the [contributing guide](https://github.com/rsuite/rsuite/blob/main/CONTRIBUTING.md#development). Feel free to ask any further questions!
